### PR TITLE
QueryControls: improve onCategoryChange prop docs in README

### PR DESCRIPTION
**Pull request by ciampo**
Quick doc tweaks as a follow-up to #46721

### CODE REVIEW

File: packages/components/src/query-controls/README.md [hunk 1]
```diff
@@ -177,9 +177,10 @@ A function that receives the new author value. If not specified, the author cont
 -   Required: No
 -   Platform: Web
 
-#### `onCategoryChange`: `CategorySelectProps[ 'onChange' ] | FormTokenFieldProps[ 'onChange' ]`
+#### `onCategoryChange`: `( newCategory: string ) => void | FormTokenFieldProps[ 'onChange' ]`
 
 A function that receives the new category value. If not specified, the category controls are not rendered.
+The function's signature changes depending on whether multiple category selection is enabled or not.
```
Comment (mirka) [inline packages/components/src/query-controls/README.md:183]: Just checking, does this line need to be added to the TS description as well?

**Review by mirka — APPROVED** [review]
🙆

**Review by kienstra — APPROVED** [review]
Hi @ciampo,
Good idea to 'expand' the `onCategoryChange` change in `README.md`

File: packages/components/src/query-controls/README.md (hunk 1 above)
Comment (ciampo) [reply packages/components/src/query-controls/README.md:183]: Good point, added in the latest commit

**Pull request merged**